ledger-tool/src/main.rs

```diff
@@ -381,7 +383,24 @@ fn output_account(
     println!("  rent_epoch: {}", account.rent_epoch());
     println!("  data_len: {}", account.data().len());
     if print_account_data {
-        println!("  data: '{}'", bs58::encode(account.data()).into_string());
+        let account_data =
+            serde_json::to_string(&UiAccount::encode(pubkey, account, encoding, None, None).data)
+                .unwrap();
+        match encoding {
+            UiAccountEncoding::Base58 => {
+                println!(" data: '{}'", bs58::encode(account.data()).into_string());
+                println!(" encoding: base58");
+            }
+            UiAccountEncoding::JsonParsed => {
+                println!("  data: '{}'", account_data);
+                println!(" encoding: jsonParsed");
+            }
+            _ => {
+                let account_and_encoding: Vec<&str> = serde_json::from_str(&account_data).unwrap();
+                println!("  data: '{}'", account_and_encoding[0]);
+                println!("  encoding: {}", account_and_encoding[1]);
+            }
+        };
```

Comment:
A couple comments here.
1. We shouldn't bother calling `UiAccount::encode` for Base58, if we're just going to encode the data directly. Let's keep this case under it's own if clause, eg:

```
if encoding == UiAccountEncoding::Base58 {
    println!(" data: '{}'", bs58::encode(account.data()).into_string());
    println!(" encoding: base58");
} else {
...
}
```

2. There's a bit of an issue with how `UiAccountEncoding::JsonParsed` is currently handled, since a call to `UiAccount::encode` with `UiAccountEncoding::JsonParsed` might actually return a `UiAccountData::Binary(data, data_encoding_str)` if the account in question is not one that solana-account-decoder knows how to parse.

What about something like this?
```
    let account_data = UiAccount::encode(pubkey, account, encoding, None, None).data;
    match account_data {
        UiAccountData::Binary(data, data_encoding_str) => {
            println!("  data: '{}'", data);
            println!("  encoding: {}", serde_json::from_str(data_encoding_str).unwrap());
        }
        UiAccountData::JsonParsed(account_data) => {
            println!("  data: '{}'", account_data);
            println!("  encoding: jsonParsed");
        }
    }
```
This also avoids the serde_json to_string back from_str, which feels clunky.

Reply:
Makes sense.

Reply:
updated

Reply:
I forgot I could use match that way. I am very new rustacean. thank you for your review